src/cpp/ext/filters/census/client_filter.cc

```diff
@@ -56,11 +56,25 @@ grpc_error_handle CensusClientCallData::Init(
 // OpenCensusCallTracer::OpenCensusCallAttemptTracer
 //
 
+OpenCensusCallTracer::OpenCensusCallAttemptTracer::OpenCensusCallAttemptTracer(
+    OpenCensusCallTracer* parent, uint64_t attempt_num,
+    bool is_transparent_retry, bool arena_allocated)
+    : parent_(parent),
+      arena_allocated_(arena_allocated),
+      context_(CensusContext(absl::StrCat("Attempt.", parent_->method_),
+                             &parent_->context_.Span(),
+                             parent_->context_.tags())),
+      start_time_(absl::Now()) {
+  GPR_DEBUG_ASSERT(parent_->context_.Context().IsValid());
```

Comment:
Prior to this PR, this assert would be run before passing the fields from `parent_->context_` to the `CensusContext` ctor, but now it's being done after.  Is that okay?

Reply:
moved to `StartNewAttempt`

Reply:
That's not quite optimal, since someone could later change the code to instantiate `OpenCensusCallAttemptTracer` from somewhere else, and they might not remember to add the assertion.

Is `CensusContext` movable?  If so, you can use a helper function like this:

```
namespace {

CensusContext CreateCensusContextForCallAttempt(
    absl::string_view method, const CensusContext& parent_context) {
  GPR_DEBUG_ASSERT(parent_context.Context().IsValid());
  return CensusContext(method, &parent_context.Span(), parent_context.tags());
}

}  // namespace
```

Then in the initializer list, you can do this:

```
       context_(CreateCensusContextForCallAttempt(
                    absl::StrCat("Attempt.", parent_->method_),
                    parent_->context_)),
```

Reply:
works!